org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -801,6 +813,14 @@ public Preferences setImportOrder(List<String> importOrder) {
 		return this;
 	}
 
+	public Preferences setFilteredTypes(List<String> filteredTypes) {
+		this.filteredTypes = (filteredTypes == null || filteredTypes.size() == 0) ? JAVA_COMPLETION_FILTERED_TYPES_DEFAULT : filteredTypes;
```

Comment:
If user decides to remove filters (sets empty list), we should not force our default filters